core/storage/trie/polkadot_trie/trie_node.hpp

```diff
@@ -132,10 +133,39 @@ namespace kagome::storage::trie {
       ReservedForCompactEncoding  // 0001 0000
     };
 
-    inline bool isBranch() const noexcept;
+    virtual bool isBranch() const noexcept = 0;
```

Comment:
Why remove `dynamic_cast`?
(`Branch` is `Leaf` with `optional` children, so they will be merged into one `TrieNode` class)